cloudinit/stages.py

```diff
@@ -406,6 +406,7 @@ def _get_data_source(self, existing) -> sources.DataSource:
                         ds,
                     )
                 else:
+                    util.del_file(self.paths.instance_link)
```

Comment:
This seems to align with previous behavior prior to  commit 9929a0058  for all non-vmware datasources. It also looks to retain VMware-specific behavior where `check_if_fallback_is_allowed == True`